lib/puppet/provider/user/useradd.rb

```diff
@@ -9,8 +9,12 @@
     install Ruby's shadow password library (often known as `ruby-libshadow`)
     if you wish to manage user passwords."
 
+  # These commands have special names for use with the base ObjectAdd provider
   commands :add => "useradd", :delete => "userdel", :modify => "usermod", :password => "chage"
 
+  # This is used in our local implementation and has no special name
+  commands :gpasswd => "gpasswd"
```

Comment:
AFAIK gpasswd should exist on all systems that have useradd/usermod, but I'll go through some really new and really old ones to see if it's an optional binary anywhere.

If so I can fixup `groups=` to fall back to the old usermod behavior if it doesn't exist (with the caveat that the bug will still exist in that case)